### PR TITLE
feat: add Min/MaxStateRootElements configurations

### PR DESCRIPTION
**Pull request by cfromknecht**
**Description**
This PR adds configuration hooks to bound the number of state roots
that are permitted in state root batches. Specifically, the minimum
allows us to amortized the cost of adding state roots to the CTC
contract. Additionally, the maximum is used in place of the old limit
based on tx size, since this is more natural and mirrors the minimum
bound.

The maximum possible value should be less than 3750 to keep in line with
our conservative 120kb limit.

**Metadata**
- Fixes ENG-2070

### CODE REVIEW

**Review by mslipper — APPROVED** [review]

**Review by mslipper — CHANGES_REQUESTED** [review]
I think there need to be some changes to the docker compose file - tests are stalling.

File: go/batch-submitter/config.go [hunk 1]
```diff
@@ -74,14 +74,18 @@ type Config struct {
...
+	// can be submitted in single proposer batch.
+	MaxStateRootElements uint64
+
 	// MaxTxBatchCount is the maximum number of L2 transactions that can ever be
 	// in a batch.
 	MaxTxBatchCount uint64
 
-	// MaxStateBatchCount is the maximum number of L2 state roots that can ever
-	// be in a batch.
-	MaxStateBatchCount uint64
```
Comment (tynes) [inline go/batch-submitter/config.go:83]: If I understand correctly, this was unused previously

**Comment by tynes** [comment]
> I think there need to be some changes to the docker compose file - tests are stalling.

Perhaps because the new config is marked as required

File: go/batch-submitter/flags/flags.go [hunk 2]
```diff
@@ -66,6 +66,20 @@ var (
 		Required: true,
 		EnvVar:   prefixEnvVar("MAX_L1_TX_SIZE"),
 	}
+	MinStateRootElementsFlag = cli.Uint64Flag{
+		Name: "min-state-root-elements",
+		Usage: "Minimum number of elements required to submit a state " +
+			"root batch",
+		Required: true,
```
Comment (tynes) [inline go/batch-submitter/flags/flags.go:73]: We could put a default of 1 here for simplicity

**Comment by tynes** [comment]
Generally looks good to me besides the need to update `ops` with the new config

File: go/batch-submitter/config.go (hunk 1 above)
Comment (cfromknecht) [reply go/batch-submitter/config.go:83]: yes it was, the member existed but it wasn't read/written anywhere

File: go/batch-submitter/flags/flags.go (hunk 2 above)
Comment (cfromknecht) [reply go/batch-submitter/flags/flags.go:73]: Ended up leaving it as optional, as this mirrors the min/max tx size

**Comment by cfromknecht** [comment]
> Perhaps because the new config is marked as required

Indeed this was the problem, thanks!

**Review by mslipper — APPROVED** [review]

**Pull request merged**